sstables/storage.cc

```diff
@@ -683,11 +683,17 @@ future<> s3_storage::snapshot(const sstable& sst, sstring dir, absolute_path abs
 
 std::unique_ptr<sstables::storage> make_storage(sstables_manager& manager, const data_dictionary::storage_options& s_opts, sstring dir, sstable_state state) {
     return std::visit(overloaded_functor {
-        [dir, state] (const data_dictionary::storage_options::local& loc) mutable -> std::unique_ptr<sstables::storage> {
-            return std::make_unique<sstables::filesystem_storage>(std::move(dir), state);
+        [state] (const data_dictionary::storage_options::local& loc) mutable -> std::unique_ptr<sstables::storage> {
+            if (loc.dir.empty()) {
+                on_internal_error(sstlog, "Non-table local storage options");
```

Comment:
This message (and the one below) is quite cryptic. Better: "local storage option is missing 'dir'".